### ISSUE TITLE
Add a preferred printing feature to Xmage deckbuilder

### ISSUE THREAD

**Issue by Svyatoslav28597** [magefree/mage#12628]
If you create your decks outside of xmage (On scryfall or Moxfield, as an exaple) and load them into xmage - you get a wild mess of images in your deck, some secret lairs, etched foils, chinese prints, normal cards, etc. And that happens every time you load your deck, so setting images you like won't help, especially if you change a lot of cards and update frequently
It would be nice if you could set each card to use specific printing every time it shows up in your deck after loading, in draft, everywhere

**Comment by tiera3** [comment]
I don't know what exports are like from moxfield, but if you can export with set information (MTGA style) and use the import feature in the deck editor, it preserves the selected printings.

**Comment by xenohedron** [comment]
Yep, the request is for a feature that xmage supports already. The described issue probably has to do with the method of import/export, or support for specifying printings from the third party service. To move forward we'd need a more detailed description of exact steps to reproduce.

**Comment by siyiye** [comment]
> If you create your decks outside of xmage (On scryfall or Moxfield, as an exaple) and load them into xmage - you get a wild mess of images in your deck, some secret lairs, etched foils, chinese prints, normal cards, etc. And that happens every time you load your deck, so setting images you like won't help, especially if you change a lot of cards and update frequently It would be nice if you could set each card to use specific printing every time it shows up in your deck after loading, in draft, everywhere

I understand what you mean: players can customize their favorite illustration version instead of randomly specifying it when importing.
For example: I only like the new version of the large-screen tron ​​picture

**Comment by Svyatoslav28597** [comment]
I usually do decks in Moxfield and import them to xmage. Moxfield offers 3 ways to export (Copy For Moxfield/Copy for MTGA/Copy for MTGO) and only copy for MTGO correctly imports to Xmage, since other two options load cards to sideboard or don't load them at all. While copy for MTGO correctly imports decks - the printings seem to be absolutely random and with each deck update on moxfield - you import cards again and see wierd bunch of secret lairs and chinese cards. Not sure if this is a quirk of moxfield, or I am missing something which solves this issue

**Comment by JayDi85** [comment]
For info: moxfield not fully support mtga format (export without set info). More details here #12304

**Comment by PurpleCrowbar** [comment]
Card import preferences could be stored in key/val pairs where the key is the card's name and the value is a string containing the set and card number separated by some delimiter e.g. `{"Go for the Throat", "40K,201"}` would reference [this specific card printing](https://scryfall.com/card/40k/201/go-for-the-throat).

Would it be fine to store these as preferences, maybe a new key in the registry inside `HKEY_CURRENT_USER\SOFTWARE\JavaSoft\Prefs\mage\client` called "importPreferences" which would contain all the pairs?

**Comment by JayDi85** [comment]
It's not about import only -- it's about deck edition too. There are already exists feature to make cards oldest and non-promo:
![shot_240905_040619](https://github.com/user-attachments/assets/f6472862-45ce-4a99-9e32-f4f4869d560b)
![shot_240905_041522](https://github.com/user-attachments/assets/c51f9c4a-1898-48bb-aa71-2826ae3dad1b)

So:
* it's a "favorite cards" feature;
* must be optional and used on import and deck editing (or only on deck editing after import to simplify GUI code -- see screen above);
* possible commands (see right click menu):
  * add selected cards to favorite list;
  * remove selected cards from favorite list;
  * replace selected/all cards by favorite list:
    * use existing favorite list;
    * if card has multiple prints then choose random versions;
  * replace selected cards by chosen set:
    * pick choice dialog with all sets list and search, see DownloadImagesDialog for example;
* add/remove to favorite feature works anytime;
* replace feature works in deck editor in free mode only;
* data stored in map/json format, search "saveSplitterLocationsToSettings" and "loadSplitterLocationsFromSettings" for example;

**Comment by JayDi85** [comment]
Moxfield and archidekt support improved in f9fd049eceb1c3650829db0fdd149dc013108e71 and it's possible to import all images now. So that's feature is not critical for deck editor now.
